File: addOne.java
import java.util.ArrayList;

public class addOne {

    public static ArrayList<Integer> oneadd(int[] digits){
        StringBuilder number=new StringBuilder();
        for (int digit:digits){
            number.append(digit);
        }
        int num=Integer.parseInt(number.toString())+1;

        ArrayList<Integer> result=new ArrayList<>();
        for(char c: String.valueOf(num).toCharArray()){
            result.add(c-'0');//accii value 49-48, 50-48, 51-48=1,2,3
        }
        return result;
    }
    
    public static void main(String[] args) {
        int[] inputDigits = {1, 2, 3};
        ArrayList<Integer> result = oneadd(inputDigits);
        System.out.println("Output: " + result);
    }
}

File: addOne.py
def add_one(digits):
    # Convert the list of digits to a number
    num = int(''.join(map(str, digits)))
    # Add 1 to the number
    num += 1
    # Convert the result back to a list of digits
    return [int(digit) for digit in str(num)]

# Example usage
input_digits = [1, 2, 3]
result = add_one(input_digits)
print("Output:", result)  # Output: [1, 2, 4]
